lib/tasks/client.rake

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+namespace :client do
+  desc "Add a new client to the bulletin board"
+
+  task :add_authority, [:name, :public_key] => :environment do |_t, args|
+    name = args[:name]
+    public_key = args[:public_key]
+    api_key = ""
+
+    existent_auth = Authority.where("name = ? or public_key = ?", name, public_key).first
+
+    if existent_auth
+      puts "The authority already exists!"
+      api_key = existent_auth.api_key
+    else
+      api_key = generate_api_key
+      Authority.create!(name: name, public_key: public_key, api_key: api_key)
+      puts "Authority '#{name}' successfuly added!"
+    end
+    puts "The API key for this authority is:"
+    puts api_key.to_s
```

Comment:
Could this be replaced with a [`find_or_initialize_by`](https://apidock.com/rails/v6.0.0/ActiveRecord/Relation/find_or_initialize_by)?

```rb
auth = Authority.find_or_initialize_by(name: name, public_key: public_key)

if auth.new_record?
  auth.update!(api_key: generate_api_key)
  puts "Authority '#{auth.name}' successfuly added!"
else
  puts "The authority already exists!"
end

puts "The API key for this authority is: #{auth.api_key.to_s}"
```

Reply:
I've done kind of a mixed solution here because I needed to ask with an OR and that's not possible with the `find_or_initialize_by` (at least I didn't find how)